notifications/push/fcm/notification.go

```diff
@@ -25,30 +26,35 @@ func NewNotification(key string) NotificationConstructor {
 		client := fcm.NewFcmClient(key).
 			SetDelayWhileIdle(true).
 			SetContentAvailable(true).
+			SetPriority(fcm.Priority_HIGH).
 			SetTimeToLive(fcm.MAX_TTL)
 
 		return &Notification{client}
 	}
 }
 
-// Send send to the tokens list.
-func (n *Notification) Send(body string, payload fcm.NotificationPayload, tokens ...string) error {
-	data := map[string]string{
-		"msg": body,
-	}
+// Send sends a push notification to the tokens list.
+func (n *Notification) Send(dataPayloadJSON string, tokens ...string) error {
+	log.Debug("calling Firebase Send", "dataPayloadJSON", dataPayloadJSON)
 
-	if payload.Title == "" {
-		payload.Title = "Status"
-	}
-	if payload.Body == "" {
-		payload.Body = "You have a new message"
+	var dataPayload map[string]string
+	err := json.Unmarshal([]byte(dataPayloadJSON), &dataPayload)
+	if err != nil {
+		return err
 	}
 
-	fmt.Println(payload.Title, payload.Body)
-
-	n.client.NewFcmRegIdsMsg(tokens, data)
-	n.client.SetNotificationPayload(&payload)
-	_, err := n.client.Send()
+	n.client.NewFcmRegIdsMsg(tokens, dataPayload)
+	resp, err := n.client.Send()
+	if err == nil {
```

Comment:
nil will be returned on L59. you need either to remove error instanation of the new err variable on L49 or restructure the code.

i suggest to follow common structure, if err != nil {return err} and then marshal json

```go
var err error
if err == nil {
	err := errors.New("1")
	fmt.Println(err)
}
fmt.Println(err)

1  
<nil>
```

Reply:
I just realized I forgot to clean up this code that I used for debugging why the notifications weren't reaching the react side. I'll do that now, sorry about that.